subsys/net/lib/nrf_cloud/src/nrf_cloud_codec_internal.c

```diff
@@ -2666,6 +2666,9 @@ int nrf_cloud_wifi_req_json_encode(struct wifi_scan_info const *const wifi,
 	int encoded_cnt = 0;
 	cJSON *wifi_obj = NULL;
 	cJSON *ap_array = NULL;
+	const bool add_all = IS_ENABLED(CONFIG_NRF_CLOUD_WIFI_LOCATION_ENCODE_OPT_ALL);
```

Comment:
Mainly for readability, I'd recommend splitting this into `add_ssid` and `add_channel`, both of which being set to `IS_ENABLED(CONFIG_NRF_CLOUD_WIFI_LOCATION_ENCODE_OPT_ALL)`.

I think that would make the logic structure a bit cleaner, and be more maintainable in the long run